programs/steward/src/state/steward_state.rs

```diff
@@ -34,26 +34,34 @@ pub struct StateTransition {
 }
 
 pub fn maybe_transition_and_emit(
-    state_account: &mut StewardState,
+    steward_state: &mut StewardState,
     clock: &Clock,
     params: &Parameters,
     epoch_schedule: &EpochSchedule,
 ) -> Result<()> {
-    let initial_state = state_account.state_tag.to_string();
-    state_account.transition(clock, params, epoch_schedule)?;
-    if initial_state != state_account.state_tag.to_string() {
+    require!(
+        clock.epoch == steward_state.current_epoch,
+        StewardError::EpochMaintenanceNotComplete
+    );
```

Comment:
Not sure if this reverted but the epoch check should be at the top of the state machine instructions https://github.com/jito-foundation/stakenet/pull/47#discussion_r1635544001

Reply:
I did this so we automatically ran the check everytime we ran a state machine function. Else, I'd have to put the check on each state machine function - which would be easy to forget to add the check